src/Storages/StorageURL.cpp

```diff
@@ -255,24 +262,94 @@ namespace
+                                /* use_external_buffer */ false,
+                                /* skip_url_not_found_error */ skip_url_not_found_error);
+
+                            Poco::Net::HTTPResponse res;
+                            buffer.call(res, Poco::Net::HTTPRequest::HTTP_HEAD);
+                            // to check if Range header is supported, we need to send a request with it set
+                            const bool supports_ranges = res.has("Accept-Ranges") && res.get("Accept-Ranges") == "bytes";
+                            LOG_TRACE(
+                                &Poco::Logger::get("StorageURLSource"),
+                                fmt::runtime(supports_ranges ? "HTTP Range is supported" : "HTTP Range is not supported"));
+
+
+                            if (supports_ranges && res.hasContentLength())
+                            {
+                                LOG_TRACE(
+                                    &Poco::Logger::get("StorageURLSource"),
+                                    "Using ParallelReadBuffer with {} workers with chunks of {} bytes",
+                                    download_threads,
+                                    settings.max_download_buffer_size);
+
+                                auto read_buffer_factory = std::make_unique<RangedReadWriteBufferFromHTTPFactory>(
+                                    res.getContentLength(),
+                                    settings.max_download_buffer_size,
+                                    request_uri,
+                                    http_method,
+                                    callback,
+                                    timeouts,
+                                    credentials,
+                                    settings.max_http_get_redirects,
+                                    DBMS_DEFAULT_BUFFER_SIZE,
+                                    read_settings,
+                                    headers,
+                                    context->getRemoteHostFilter(),
+                                    delay_initialization,
+                                    /* use_external_buffer */ false,
+                                    /* skip_url_not_found_error */ skip_url_not_found_error);
+                                return wrapReadBufferWithCompressionMethod(
+                                    std::make_unique<ParallelReadBuffer>(
+                                        std::move(read_buffer_factory), &IOThreadPool::get(), download_threads),
+                                    chooseCompressionMethod(request_uri.getPath(), compression_method));
+                            }
+                        }
+                        catch (...)
+                        {
+                            LOG_TRACE(
+                                &Poco::Logger::get(__PRETTY_FUNCTION__),
+                                "Failed to setup ParallelReadBuffer. Falling back to the single-threaded buffer");
```

Comment:
LOG_ERROR? Also may be print exception message?

Reply:
`LOG_ERROR` seems a bit too much as this is only checking if it's possible to use ParallelReadBuffer. 
For example, exception can happen if `HEAD` is not supported. While this error disables `ParallelReadBuffer` it still allows us to try and do `GET`.
Maybe something like `LOG_WARNING` instead?

Reply:
ah you are right, then both TRACE or WARNING are ok, but adding exception message to log in this place is anyway helpful.

Reply:
though we could also check exact http error codes here, like HTTP_METHOD_NOT_ALLOWED, but not necessary I think